tap_learndash/streams.py

```diff
@@ -1,46 +1,755 @@
+        API.
+        """
+        row["user_id"] = context["user_id"]
+        row["course_id"] = context["course_id"]
+        return row
+
+
+class UserCoursesStream(LearnDashStream):
+    """Define custom stream."""
+    name = "user_courses"
+    path = "/users/{user_id}/courses"
+    primary_keys = ["user_id", "id"]
+    parent_stream_type = CoursesUsersStream
+    ignore_parent_replication_keys = True
+    schema = th.PropertiesList(
+        th.Property("user_id", th.IntegerType),
+        th.Property("id", th.IntegerType),
+        th.Property("date", th.DateTimeType),
+        th.Property("date_gmt", th.DateTimeType),
+        th.Property("guid", th.ObjectType(
+            th.Property("rendered", th.StringType)
+        )),
+        th.Property("modified", th.DateTimeType),
+        th.Property("modified_gmt", th.DateTimeType),
+        th.Property("slug", th.StringType),
+        th.Property("status", th.StringType),
+        th.Property("type", th.StringType),
+        th.Property("link", th.StringType),
+        th.Property("title", th.ObjectType(
+            th.Property("rendered", th.StringType)
+        )),
+        th.Property("content", th.ObjectType(
+            th.Property("protected", th.BooleanType),
+            th.Property("rendered", th.StringType)
+        )),
+        th.Property("author", th.IntegerType),
+        th.Property("featured_media", th.IntegerType),
+        th.Property("menu_order", th.IntegerType),
+        th.Property("template", th.StringType),
+        th.Property("categories", th.ArrayType(th.StringType)),
+        th.Property("tags", th.ArrayType(th.StringType)),
+        th.Property("ld_course_category", th.ArrayType(th.IntegerType)),
+        th.Property("ld_course_tag", th.ArrayType(th.StringType))
+    ).to_dict()
+
+    def post_process(self, row: dict, context: Optional[dict] = None) -> dict:
+        """As needed, append or transform raw data to match expected structure.
+        Optional. This method gives developers an opportunity to "clean up" the results
+        prior to returning records to the downstream tap - for instance: cleaning,
+        renaming, or appending properties to the raw record result returned from the
+        API.
+        """
+        row["user_id"] = context["user_id"]
+        return row
+
+
+class UserGroupsStream(LearnDashStream):
+    """Define custom stream."""
+    # At the moment this stream returns empty so I have no good idea what to add on the properties list
+    name = "user_groups"
+    path = "/users/{user_id}/groups"
+    primary_keys = ["user_id", "id"]
+    parent_stream_type = CoursesUsersStream
+    ignore_parent_replication_keys = True
+    schema = th.PropertiesList(
+        th.Property("user_id", th.IntegerType)
+    ).to_dict()
+
+
+class UserQuizProgressStream(LearnDashStream):
+    """Define custom stream."""
+    # At the moment this stream returns empty so I have no good idea what to add on the properties list
```

Comment:
Maybe we should put all the `quiz` components at the bottom and comment them out?